src/tlo/methods/schisto.py

```diff
@@ -333,18 +677,60 @@ def do_at_generic_first_appt(
         person_id: int,
         symptoms: List[str],
         schedule_hsi_event: HSIEventScheduler,
-        **kwargs
+        **kwargs,
     ) -> None:
         # Do when person presents to the GenericFirstAppt.
         # If the person has certain set of symptoms, refer ta HSI for testing.
-        set_of_symptoms_indicative_of_schisto = {'anemia', 'haematuria', 'bladder_pathology'}
-
-        if set_of_symptoms_indicative_of_schisto.issubset(symptoms):
+        set_of_symptoms_indicative_of_schisto = {'anemia',
+                                                 'haematuria',
+                                                 'bladder_pathology',
+                                                 'fever',
+                                                 'ascites',
+                                                 'diarrhoea',
+                                                 'vomiting',
+                                                 'hepatomegaly',
+                                                 'dysuria'}
+
+        if any(symptom in set_of_symptoms_indicative_of_schisto for symptom in symptoms):
             event = HSI_Schisto_TestingFollowingSymptoms(
                 module=self, person_id=person_id
             )
             schedule_hsi_event(event, priority=0, topen=self.sim.date)
 
+    def select_test(self, person_id):
+
+        # choose test
+        persons_symptoms = self.sim.modules["SymptomManager"].has_what(person_id)
+        if any(symptom in ['haematuria', 'bladder_pathology',] for symptom in persons_symptoms):
```

Comment:
as above, as we're dealing with sets naturally it's faster to do the subset operation than iteration.